httplogger/httplogger.go

```diff
@@ -109,35 +112,40 @@ func LoggingMiddlewareLogrus(logger *logrus.Entry, next http.Handler) http.Handl
 
 // LoggingMiddlewareZap logs the incoming HTTP request & its duration.
 func LoggingMiddlewareZap(logger *zap.SugaredLogger, next http.Handler) http.Handler {
-	return http.HandlerFunc(
-		func(w http.ResponseWriter, r *http.Request) {
-			defer func() {
-				if err := recover(); err != nil {
-					w.WriteHeader(http.StatusInternalServerError)
-
-					method := ""
-					url := ""
-					if r != nil {
-						method = r.Method
-						url = r.URL.EscapedPath()
-					}
-
-					logger.With(
-						"err", err,
-						"trace", string(debug.Stack()),
-						"method", r.Method,
-					).Errorf("http request panic: %s %s", method, url)
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// Generate request ID (`base64` to shorten its string representation)
+		_uuid := [16]byte(uuid.New())
+		httpRequestID := base64.RawStdEncoding.EncodeToString(_uuid[:])
```

Comment:
why base64 on top? maybe better make md5 to shorten the uid?

Reply:
```
❯ go run main.go -log-dev  

2023-06-27T13:24:18.733+0200    INFO    httplogger/main.go:44   Webserver running at localhost:8124
2023/06/27 13:24:24 http: superfluous response.WriteHeader call from main.HelloHandler (main.go:20)
2023-06-27T13:24:24.155+0200    INFO    httplogger/httplogger.go:146    Handled HTTP request: GET /hello 0      {"httpRequestID": "8lDF96hSQOKY/7mlvKKU9g", "duration": "55µs", "status": 0, "method": "GET", "path": "/hello"}
```

Reply:
nice! thxxx